### PR TITLE
Audit fixes

### PR DESCRIPTION
**Pull request by c0rv0s**
Addresses the concerns raised by the audit.

Notes:
- QSP-4 suggests using an Enumerable Map data structure but I'm not sure that really makes sense here since the Open Zeppelin implementation only supports `uint -> address`
- QSP-9 is solved by the solution to QSP-2 which prevents duplicate memberId stakes and duplicate payrollIds

### CODE REVIEW

**Review by Niraj-Kamdar — APPROVED** [review]
LGTM! Just wondering why we are using require at one place instead of Error

File: contracts/OpolisPay.sol [hunk 1]
```diff
@@ -126,22 +135,24 @@ contract OpolisPay {
...
         ) revert InvalidStake();
         if (memberId == 0) revert NotMember();
+        if (stakes[memberId]) revert DuplicateStake();
         
         // @dev function for auto transfering out stakes 
 
         if (msg.value > 0 && token == address(0)){
-            destination.transfer(msg.value);
+            (bool success, ) = destination.call{value: msg.value}("");
+            require(success, "Transfer failed.");
```
[the remark this answers is not in the document]
Comment (c0rv0s) [reply contracts/OpolisPay.sol:151]: ah i just copied this line from audit report

File: contracts/OpolisPay.sol [hunk 2]
```diff
@@ -41,27 +49,27 @@ error DirectTransfer();
...
-    address public opolisAdmin; //Should be Opolis multi-sig for security
-    address payable public destination; // Where funds are liquidated 
-    address public opolisHelper; //Can be bot wallet for convenience 
-    
-    uint256[] public payrollIds; //List of payrollIds associated with payments
-    uint256[] public stakes; //List of members who have staked
+contract OpolisPay is ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    address[] private supportedTokens; //Tokens that can be sent.
```
Comment (wswarren12) [inline contracts/OpolisPay.sol:55]: I think we should make this a public variable.

File: contracts/OpolisPay.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (wswarren12) [reply contracts/OpolisPay.sol:151]: Yeah, let's fix that too, since it's a small change.

**Review by wswarren12 — APPROVED** [review]
Left one more minor comment, but otherwise looks great.

**Pull request merged**